microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsMpService.java

```diff
@@ -46,22 +45,12 @@ public void configure(MpServiceContext serviceContext) {
 
         MetricsSupport metricsSupport = MetricsSupport.create(metricsConfig);
 
-        ConfigValue<String> routingNameConfig = metricsConfig.get("routing").asString();
-        Routing.Builder defaultRouting = serviceContext.serverRoutingBuilder();
+        RoutingBuilders routingBuilders = RoutingBuilders.newRoutingBuilders(
+                serviceContext, metricsConfig);
 
-        Routing.Builder endpointRouting = defaultRouting;
-
-        if (routingNameConfig.isPresent()) {
-            String routingName = routingNameConfig.get();
-            // support for overriding this back to default routing using config
-            if (!"@default".equals(routingName)) {
-                endpointRouting = serviceContext.serverNamedRoutingBuilder(routingName);
-            }
-        }
-
-        metricsSupport.configureVendorMetrics(null, defaultRouting);
+        metricsSupport.configureVendorMetrics(null, routingBuilders.defaultBuilder());
```

Comment:
The `null` as the first parameter is terrible, though I did that :(

Reply:
I stand on the shoulders of giants...